bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/regexp/TreeBuilder.java

```diff
@@ -292,7 +404,21 @@ private String readCharacterClassEnd() {
             Token consumedToken = consume();
             return consumedToken.value;
         }
-        throw new BallerinaException("Missing ']' character");
+        throw new BallerinaException(BLangExceptionHelper.getErrorMessage(
+                RuntimeErrors.REGEXP_MISSING_CLOSE_BRACKET.messageKey()));
+    }
+
+    private RegExpQuantifier readOptionalQuantifier() {
+        Token nextToken = peek();
+        switch (nextToken.kind) {
+            case QUESTION_MARK_TOKEN:
+            case ASTERISK_TOKEN:
+            case PLUS_TOKEN:
+            case OPEN_BRACE_TOKEN:
+                return readReQuantifier();
+            default:
+                return new RegExpQuantifier("", "");
```

Comment:
Why we have to return a default case here?

Reply:
For cases where we do not have quantifier.

The return of this method is used to create a `ReAtomQuantifier`. We need to have a empty string if there is no quantifier present. Returning a null won't help. So that I have return `RegExpQuanifier` with empty value.

Reply:
Can't we return null in here and handle the null value, in the upper layers

Reply:
Yes, we can. I'll change.